js/apps/admin-ui/test/realm-settings/oid4vci-attributes.spec.ts

```diff
@@ -3,107 +3,111 @@ import { v4 as uuid } from "uuid";
 import adminClient from "../utils/AdminClient";
 import { login } from "../utils/login";
 import { goToRealm, goToRealmSettings } from "../utils/sidebar";
-import { assertAxeViolations } from "../utils/masthead";
 
-test.describe("OID4VCI Attributes tab", () => {
-  const realmName = `oid4vci-realm-${uuid()}`;
-
-  test.beforeAll(async () => {
-    await adminClient.createRealm(realmName);
-  });
-
-  test.afterAll(async () => {
-    await adminClient.deleteRealm(realmName);
-  });
-
-  test.beforeEach(async ({ page }) => {
-    await login(page);
-    await goToRealm(page, realmName);
-    await goToRealmSettings(page);
-  });
-
-  test("should handle OID4VCI tab visibility based on feature flag", async ({
-    page,
-  }) => {
-    const oid4vciTab = page.getByTestId("rs-oid4vci-attributes-tab");
-    const isVisible = await oid4vciTab.isVisible();
-
-    if (isVisible) {
-      await expect(oid4vciTab).toBeVisible();
-    } else {
-      await expect(oid4vciTab).toBeHidden();
-    }
-  });
-
-  test("should render fields and save values with correct attribute keys", async ({
-    page,
-  }) => {
-    const oid4vciTab = page.getByTestId("rs-oid4vci-attributes-tab");
-    const isVisible = await oid4vciTab.isVisible();
+const realmName = `oid4vci-realm-${uuid()}`;
+let isFeatureEnabled: boolean = true;
+
+test.beforeAll(async ({ browser }) => {
+  await adminClient.createRealm(realmName);
+  const realm = await adminClient.getRealm(realmName);
+  expect(realm).toBeDefined();
+
+  // Check if feature is enabled using a fresh context
+  const page = await browser.newPage();
+  await login(page);
+  await goToRealm(page, realmName);
+  await goToRealmSettings(page);
+  const oid4vciTab = page.getByTestId("rs-oid4vci-attributes-tab");
+  await page.waitForTimeout(1000);
+  isFeatureEnabled = (await oid4vciTab.count()) > 0;
+  await page.close();
+});
 
-    if (isVisible) {
-      await oid4vciTab.click();
+test.afterAll(async () => {
+  await adminClient.deleteRealm(realmName);
+});
 
-      const nonceField = page.getByTestId("oid4vci-nonce-lifetime-seconds");
-      const preAuthField = page.getByTestId("pre-authorized-code-lifespan-s");
+test("OID4VCI tab visibility", async ({ page }) => {
+  await login(page);
+  await goToRealm(page, realmName);
+  await goToRealmSettings(page);
+  await page.waitForTimeout(1000);
+
+  const oid4vciTab = page.getByTestId("rs-oid4vci-attributes-tab");
+
+  if (!isFeatureEnabled) {
+    // When feature is disabled, verify tab is NOT present
+    await expect(oid4vciTab).toHaveCount(0);
+  } else {
+    // When feature is enabled, verify tab is visible
+    await expect(oid4vciTab).toBeVisible();
+  }
```

Comment:
This update does not quite address the issue I previously raised about conditional branching in tests. Simply reading, I still don't know which branch to take and what is the behavior that you expect. 

I guess this whole testsuite does not matter if the OID4VCI feature is not on. Isn't that explicitly enabled in test configs? Edit: I checked and it is the case: 
https://github.com/adorsys/keycloak-oid4vc/blob/71f510d115a34ac78d56c1efd4cf5c4a9e9ef831/js/apps/keycloak-server/scripts/start-server.js#L61-L65

Then I suggest to simply assert that the tab is visible.

```suggestion
await expect(oid4vciTab).toBeVisible();
```
But you seem reluctant to. Could you communicate your rationale?